Modules/Common/AptosLikeMainModule.php

```diff
@@ -0,0 +1,297 @@
+                            if ($diff === '0')
+                            {
+                                continue;
+                            }
+
+                            $diff_sum = bcadd($diff_sum, $diff);
+                            $events[] = [
+                                'block' => $block['block_height'],
+                                'transaction' => $trx['hash'],
+                                'time' => $this->block_time,
+                                'address' => $address,
+                                'sort_key' => $sort_key++,
+                                'effect' => $diff,
+                                'failed' => $failed,
+                                'extra' => null,
+                            ];
+                        }
+
+                        if ($diff_sum !== '0')
+                        {
+                            $events[] = [
+                                'block' => $block['block_height'],
+                                'transaction' => $trx['hash'],
+                                'time' => $this->block_time,
+                                'address' => 'the-contract',
+                                'sort_key' => $sort_key++,
+                                'effect' => bcmul($diff_sum, '-1'),
+                                'failed' => $failed,
+                                'extra' => null,
+                            ];
+                        }
+                    }
+
+                    // Process transaction Fee
+                    $events[] = [
+                        'block' => $block['block_height'],
+                        'transaction' => $trx['hash'],
+                        'time' => $this->block_time,
+                        'address' => $trx['sender'],
+                        'sort_key' => $sort_key++,
+                        'effect' => '-' . $fee,
+                        'failed' => $failed,
+                        'extra' => 'f',
+                    ];
+
+                    $events[] = [
+                        'block' => $block['block_height'],
+                        'transaction' => $trx['hash'],
+                        'time' => $this->block_time,
+                        'address' => 'the-void',
+                        'sort_key' => $sort_key++,
+                        'effect' => $fee,
+                        'failed' => $failed,
+                        'extra' => 'f',
+                    ];
+
+                    break;
+
+                default:
+                    break;
+            }
+        }
+
+        $this->set_return_events($events);
+    }
+
+    public function api_get_balance(string $address): string
+    {
+        try
+        {
+            $resource = urlencode("0x1::coin::CoinStore<0x1::aptos_coin::AptosCoin>");
+            $result = requester_single($this->select_node(), endpoint: "v1/accounts/{$address}/resource/$resource", timeout: $this->timeout);
+        } catch (RequesterException $e)
```

Comment:
How to differ "no resource" from the API going down?

Reply:
Fixed with `valid_codes` and parsing the curl result.